lib/fastlane/actions/nexus_upload.rb

```diff
@@ -0,0 +1,154 @@
+module Fastlane
+  module Actions
+    class NexusUploadAction < Action
+      def self.run(params)
+        command = []
+        command << "curl"
+        command << verbose(params)
+        command += ssl_options(params)
+        command += proxy_options(params)
+        command += upload_options(params)
+        command << upload_url(params)
+
+        Fastlane::Actions.sh(command.join(' '), log: false)
+      end
+
+      def self.upload_url(params)
+        "#{params[:endpoint].shellescape}/nexus/service/local/artifact/maven/content"
+      end
+
+      def self.verbose(params)
+        params[:verbose] ? "--verbose" : "--silent"
+      end
+
+      def self.upload_options(params)
+        file_path = File.expand_path(params[:file]).shellescape
+        file_extension = file_path.split('.').last.shellescape
+
+        options = []
+        options << "-F p=zip"
+        options << "-F hasPom=false"
+        options << "-F r=#{params[:repo_id].shellescape}"
+        options << "-F g=#{params[:repo_group_id].shellescape}"
+        options << "-F a=#{params[:repo_project_name].shellescape}"
+        options << "-F v=#{params[:repo_project_version].shellescape}"
+        options << "-F e=#{file_extension}"
+        options << "-F file=@#{file_path}"
+        options << "-u #{params[:username].shellescape}:#{params[:password].shellescape}"
+
+        options
+      end
+
+      def self.ssl_options(params)
+        options = []
+        unless params[:ssl_verify]
```

Comment:
In some organisations Nexus will be hosted on internal servers, with self signed SSL/TLS certificate. And to be able to upload it is needed to disable ssl check.